src/application-delegate.coffee

```diff
@@ -130,26 +130,34 @@ class ApplicationDelegate
   getUserDefault: (key, type) ->
     remote.systemPreferences.getUserDefault(key, type)
 
-  confirm: ({message, detailedMessage, buttons}) ->
-    buttons ?= {}
-    if Array.isArray(buttons)
-      buttonLabels = buttons
+  confirm: (options, callback) ->
+    if callback?
+      # Async version: pass options directly to Electron but set sane defaults
+      options = Object.assign({type: 'info', normalizeAccessKeys: true}, options)
+      remote.dialog.showMessageBox(remote.getCurrentWindow(), options, callback)
     else
-      buttonLabels = Object.keys(buttons)
-
-    chosen = remote.dialog.showMessageBox(remote.getCurrentWindow(), {
-      type: 'info'
-      message: message
-      detail: detailedMessage
-      buttons: buttonLabels
-      normalizeAccessKeys: true
-    })
-
-    if Array.isArray(buttons)
-      chosen
-    else
-      callback = buttons[buttonLabels[chosen]]
-      callback?()
+      # Legacy sync version: options can only have `message`,
+      # `detailedMessage` (optional), and buttons array or object (optional)
+      {message, detailedMessage, buttons} = options
```

Comment:
It seems like it would be very unintuitive, and therefore cause difficult to debug issues, if the sync and async versions don't have the same options

Reply:
I would vote for deprecating `detailedMessage` for `detail` but keep `buttons` as an array or object

Reply:
Since the new async version is opt-in, I think that should immediately warn package authors that they need to look at their code and update it accordingly.  I explicitly didn't update the sync code path because the goal I had in mind when creating this PR was to completely deprecate and feature-freeze it to give package authors more "incentive" to moving to async.